public/app/plugins/datasource/loki/types.ts

```diff
@@ -35,6 +35,12 @@ export interface LokiQuery extends LokiQueryFromSchema {
   // the temporary fix (until this gets improved in the codegen), is to
   // override it here
   queryType?: LokiQueryType;
+
+  /**
+   * This is a property for the experimental query splitting feature.
+   * @experimental
+   */
+  experimentalChunkRange?: string;
```

Comment:
could we do the thing where this stores milliseconds-as-number, and when showing it in the `form`, we convert to-and-from a string-representation? WDYT?

(best would be those "nice" string representations like "5m", but for the initial run we could also live with just having people enter the millisecond-value)

Reply:
I went with that because I only found `parseDuration` which works `string -> Duration` and `durationToMilliseconds` for `Duration -> number`. I am not sure if we can easily parse a duration string from milliseconds. How would you do that?

I also don't have a big problem with that being a string 😜

Reply:
>  I am not sure if we can easily parse a duration string from milliseconds. How would you do that?

efficiently. 😁 


on a more serious note, i'm totally fine with it being a string for the initial version,

but i would strongly object to having it a string in the "production ready" version.. the problem is, the "stability" of that thing (in a dashboard case) depends on the "stability" of the string->number transform-function.. in other words, if that function changes, we suddenly have different values here. in this case we basically outsource the transforming to `parseDuration` and to https://date-fns.org/v2.29.3/docs/add . also, i think this cannot do miliseconds.. so if someone asks "what are the allowed formats of data in this field", we don't really know...

meh, as i said, i'm totally fine with going with string-first, but we should keep in mind that this may have to change later.

Reply:
Well, that would be a problem, but basically not just ours. Things like `timeShift` or `interval` are also saved as strings with exactly the same format:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/8092184/225567772-d8ff87af-939d-4a96-a08f-27b1325d2b11.png">

Reply:
What I would make sure is that `querySplitting` receives a numeric value, and doesn't have to worry about how it's is stored (string or anything) or presented (duration format). If we need to store it as a duration string it's fine, but I would pass it as a number, otherwise we're leaking external details into the _chunking_ code.

Reply:
(nitpick, that is `minInterval` technically 😁 )

Reply:
It's part of the `queries`/`request` object which is being passed to `querySplitting`.

Reply:
(another point is, i think we can assume we will never interpret this strings on the server, right?)

but yeah, let's string it for now 😄

Reply:
> (nitpick, that is `minInterval` technically 😁 )

Well, I just looked into a dashboard JSON and that's the first I found 🤷

Reply:
> (another point is, i think we can assume we will never interpret this strings on the server, right?)

Not sure? But even if, Duration strings are common to be interepreted on the server, like we do with our range vectors in metric queries (e.g. the `1h` in `rate({job="\"grafana/data\""} [1h])`)

Reply:
> It's part of the queries/request object which is being passed to querySplitting.

Which is an object we have control over and currently transform (fixedRequest). 

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/datasource.ts#L256-L263

Reply:
I see, so should I adjust the type extension [here](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/datasource.ts#L260) and add a number type? But that would also mean I'd need to change the type in `runPartitionedQueries`.

Can do that, but honestly I don't feel we are gaining much.

Reply:
Honestly, yes. If we could store it as a number it would fix this for us, but since it's not possible I don't think it's worth going the extra mile for this.